### PR TITLE
[tohono_oodham] Initial mobile keyboard for Tohono O'odham

### PR DESCRIPTION
**Pull request by alevy03**
This is a new keyboard to input script for the Tohono O'odham language. Only implemented for mobile platforms, currently.
This is my first time submitting a keyboard. So critiques, suggestions, and pointers are much appreciated!

### CODE REVIEW

**Comment by darcywong00** [comment]
Thanks for your keyboard submission! My suggestion is using Keyman Developer 10 to also create a Keyman package (named `release/t/tohono_oodham/source/tohono_oodham.kps`) as a way of distributing your touch layout keyboard.

The package tutorial is https://help.keyman.com/developer/10.0/guides/distribute/tutorial/

Make sure in Step 3 that you assign the BCP 47 language code `ood`.

In Step 4, you can leave the package version blank, and tick the box to have the package version follow the keyboard version.

**Comment by alevy03** [comment]
Hi,
   Thanks for looking at my pull request. I’ll make the changes you suggested and add them to the PR soon (Lord willing)

~ Alfred

**Comment by LornaSIL** [comment]
The build is still failing. The only thing I can think of is in the keyboard_info file you should change ```'``` to ```&#39;``` so instead of:
```    "description": "This keyboard is used to enter text in the language of the Tohono O'odham people."```
it would be:
```    "description": "This keyboard is used to enter text in the language of the Tohono O&#39;odham people."```

Here is one other requested change:
* In the .kmn header you should include a version and copyright line. Add the following to .kmn
```
store(&COPYRIGHT) '© 2018 James Wyly'
store(&KEYBOARDVERSION) '0.1'
```
If we can get beyond the build failing it should be ready to approve (unless you want to add Desktop support too).

**Comment by LornaSIL** [comment]
Oh, then that means you have to include the Desktop keyboard. @alevy03 

If that's the case, then I believe you also need to include screenshots of the keyboard layout in the welcome.htm file. 

You can do that by clicking on Details / Features / Add ... / Desktop On Screen Keyboard. Then, you can "Fill from layout" Hm, but I see you haven't implemented the special characters in the Desktop keyboard.

I *think* for the sake of the quality of the keyboard we'll have to ask you to add these characters to your Desktop keyboard. It would be very strange if you had a Desktop keyboard that didn't support the same characters as are in your mobile keyboard.

In the Keyman file you need to say that your keyboard supports Windows at a minimum.

You can delete the "=" rule in your keyboard and add this code:
```
 + "=" > dk(1)
dk(1) + "="   > U+003D 
dk(1) + "i"   > U+012D 
dk(1) + "s"   > U+1E63
dk(1) + "d"   > U+1E0D
dk(1) + "n"   > U+00F1
dk(1) + "m"   > U+014B 
```
That would allow you to type the keystrokes with a deadkey of "=". Of course, you could choose another key as your deadkey or you could implement it another way. Does this language not need uppercase characters to match these? If the language needs them, then you should add those too.

**Comment by LornaSIL** [comment]
@alevy03 
This is something I'm not real familiar with, but another option would just be to edit the keyboard_info file and add the platform support code so your file would look like this:
```
{
    "license": "mit",
    "jsFilename": "tohono_oodham.js",
    "platformSupport": {
        "desktopWeb": "none",
        "ios": "full",
        "android": "full"
    },
    "languages": ["ood"],
    "description": "This keyboard is used to enter text in the language of the Tohono O'odham people."
}
```
I *think* this would allow it so you don't have to add the .kmx into the package.

**Comment by darcywong00** [comment]
You can check if your changes will work on the build server using the Git Bash prompt in your keyboards directory to run the build script:

```
./build.sh release/t/tohono_oodham/
```

**Comment by darcywong00** [comment]
I tried modifying the .keyboard_info as @LornaSIL suggested, but I still got the same error. 

It might be something in kmcomp.exe to chase down (see issue 11110 above)...

**Comment by alevy03** [comment]
Thanks for all the suggestions. I have been busy with other things and haven’t had time to debug the issue. I do plan to add the uppercase keys as well as figure out why the build is failing. When I can get back to this, that is. Hopefully next week.

**Comment by darcywong00** [comment]
> as well as figure out why the build is failing.

Please be patient, as it'll be 2+ weeks before we get to fixing issue 1110. Once that clears, your keyboard should build fine on the CI server.

**Comment by darcywong00** [comment]
Now that the `kmcomp` compiler in the keyboards repo has been updated, you'll need to keep your fork in sync so your keyboard will compile.

Please refer to the section **Keep your fork of the Keyboards repository in sync** of the [keyboard submission guide](https://help.keyman.com/developer/keyboards/advanced/step-1.php) to update your `master` branch of https://github.com/alevy03/keyboards

After that, you'll merge your `master` branch into your `tohono_oodham` branch and push.

**Comment by keyman-server** [comment]
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

**Comment by keyman-server** [comment]
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

File: release/t/tohono_oodham/source/tohono_oodham.kmn [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
++ [SHIFT K_H] > 'H'
++ [SHIFT K_G] > 'G'
++ [SHIFT K_E] > 'E'
++ [SHIFT K_D] > 'D'
++ [SHIFT K_C] > 'C'
++ [SHIFT K_B] > 'B'
++ [SHIFT K_A] > 'A'
+
++ [SHIFT K_N] > 'N'
++ [K_QUOTE] > U+0027
```
Comment (darcywong00) [inline release/t/tohono_oodham/source/tohono_oodham.kmn]: lines 11-105 look like standard English (US) output, so you should be able to remove them.

File: release/t/tohono_oodham/source/tohono_oodham.kmn [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
++ [T_i_bowl_top]   > U+012D 
++ [T_s_dot_bottom] > U+1E63
++ [T_d_dot_bottom] > U+1E0D
++ [T_n_tilde_top]  > U+00F1
++ [T_n_long_leg]   > U+014B 
++ [SHIFT T_cap_D_dot_bottom] > U+1E0C
++ [SHIFT T_cap_S_dot_bottom] > U+1E62
++ [SHIFT T_cap_N_tilde_top] > U+00D1
```
Comment (darcywong00) [inline release/t/tohono_oodham/source/tohono_oodham.kmn]: Just checking that your keyboard doesn't need capitalized versions of 
* i_bowl_top
* n_long_leg

for the shift layer...

File: release/t/tohono_oodham/source/tohono_oodham.kmn (hunk 1 above)
Comment (alevy03) [reply release/t/tohono_oodham/source/tohono_oodham.kmn]: Cool. Good to know that is all redundant. I've removed those lines.

**Comment by keyman-server** [comment]
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

File: release/t/tohono_oodham/source/tohono_oodham.kmn (hunk 2 above)
Comment (alevy03) [reply release/t/tohono_oodham/source/tohono_oodham.kmn]: James Wyly, who is working with this people group, said that they only use the three capitals that I've added here. So I think this is correct.

One more thing, should I change the version number to 1.0 before it gets merged, or do you think I should leave it at 0.1?

Comment (darcywong00) [reply release/t/tohono_oodham/source/tohono_oodham.kmn]: > One more thing, should I change the version number to 1.0 before it gets merged, or do you think I should leave it at 0.1?

Initial keyboard version 1.0 is preferred. They're in the markdown (.md) files too.

**Comment by keyman-server** [comment]
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

**Comment by DavidLRowe** [comment]
@alevy03 I haven't reviewed the keyboard, but just want to double-check on the issue of capitals. At some point, might someone want to put a title in all capitals and then complain that the capital form of certain letters can't be typed? If it's not too difficult to add the capitals now, it might save more trouble later. It's fine if you decide not to add the capitals, but just wanted to let you get them into version 1.0.

**Comment by alevy03** [comment]
I am messaging my language expert to see what he thinks.

**Comment by alevy03** [comment]
I’ve got confirmation, from a language expert, that the ĭ and ŋ do not have capital forms.

**Pull request merged**